File: Sorting/QuickSort.py
def partition(arr, s, e):
    i = s
    pivot = arr[e]
    for j in range(s, e):
        if arr[j] < pivot:
            arr[i], arr[j] = arr[j], arr[i]
            i += 1
    arr[i], arr[e] = arr[e], arr[i]
    return i


def quickSort(arr, s, e):
    if (s >= e):
        return

    p = partition(arr, s, e)
    quickSort(arr, s, p-1)
    quickSort(arr, p+1, e)


array = [1000, 4, 90, 80, 20, 40, -2]

print(array)

quickSort(array, 0, len(array)-1)

print(array)

File: Sorting/QuickSort.cpp
#include <iostream>
using namespace std;

int partition(int arr[], int s, int e){
    int i = s;
    int pivot = arr[e];
    for(int j = s; j<e; j++){
        if (arr[j] < pivot){
            swap(arr[i],arr[j]);
            i++;
        }
    }
    swap(arr[i],arr[e]);
    return i;
}


void QuickSort(int arr[], int s, int e){
    if (s>=e){
        return;
    }
    int p = partition(arr,s,e);
    QuickSort(arr,s,p-1);
    QuickSort(arr,p+1,e);
}

int main(){
    int arr[] = {1000, 4, 90, 80, 20, 40, -2};
    for(int i = 0; i<7;i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl;

    QuickSort(arr,0, 7);

    for(int i = 0; i<7;i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl;
    return 0;
}